util/git/creds_test.go

```diff
@@ -0,0 +1,122 @@
+package git
+
+import (
+	"github.com/argoproj/argo-cd/v2/util/io"
+	"github.com/google/uuid"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"io/ioutil"
+	"os"
+	"strings"
+	"testing"
+)
+
+type cred struct {
+	username string
+	password string
+}
+
+type memoryCredsStore struct {
+	creds map[string]cred
+}
+
+func (s *memoryCredsStore) Add(username string, password string) string {
+	id := uuid.New().String()
+	s.creds[id] = cred{
+		username: username,
+		password: password,
+	}
+	return id
+}
+
+func (s *memoryCredsStore) Remove(id string) {
+	delete(s.creds, id)
+}
+
+func TestHTTPSCreds_Environ_no_cert_cleanup(t *testing.T) {
+	store := &memoryCredsStore{creds: map[string]cred{}}
+	creds := NewHTTPSCreds("", "", "", "", true, "", store)
+	closer, env, err := creds.Environ()
+	require.NoError(t, err)
+	nonce := ""
+	for _, envVar := range env {
+		if strings.HasPrefix(envVar, ASKPASS_NONCE_ENV) {
+			nonce = envVar[len(ASKPASS_NONCE_ENV) + 1:]
+			break
+		}
+	}
+	assert.Contains(t, store.creds, nonce)
+	io.Close(closer)
+	assert.NotContains(t, store.creds, nonce)
+}
+
+func TestHTTPSCreds_Environ_insecure_true(t *testing.T) {
+	store := &memoryCredsStore{creds: map[string]cred{}}
```

Comment:
m unsure, but shouldn't there be make used?

Reply:
Googled around, looks like it's up to preference. Switching to `make`.